saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/OpenSamlAuthenticationProvider.java

```diff
@@ -122,7 +123,7 @@
  * @see <a href="https://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf#page=38">SAML 2 StatusResponse</a>
  * @see <a href="https://wiki.shibboleth.net/confluence/display/OS30/Home">OpenSAML 3</a>
  */
-public final class OpenSamlAuthenticationProvider implements AuthenticationProvider {
+public final class OpenSamlAuthenticationProvider implements AuthenticationManager, AuthenticationProvider {
```

Comment:
This might be a breaking change. For example, if users have exposed `OpenSamlAuthenticationProvider` as a bean in 5.2.x, e.g.:

```java
@Bean
public OpenSamlAuthenticationProvider provider() {
    // ... 
}
```

Then startup may fail when upgrading to 5.3.x as there will be more than one instance of `AuthenticationManager` in the application context.

You can see related behavior by changing `public interface AuthenticationProvider` to `public interface AuthenticationProvider extends AuthenticationManager` and running `MethodSecurityInterceptorWithAopConfigTests`. It's not the same thing, but it may give you some places to poke around in order to get a better idea of the implications of the change.

Reply:
That's a good point. I will revert that change.

Reply:
That's a good point. I will remove the redundant interface